edgedb-protocol/src/model/time.rs

```diff
@@ -98,6 +104,335 @@ impl Duration {
+                    let hour = hour_str
+                        .parse::<i32>()
+                        .map_err(|e| ParseDurationError::from(e)
+                            .pos(pos)
+                        )?;
+                    result += (hour as i64) * MICROS_PER_HOUR;
+                    pos += part.len();
+                    current = parts.next();
+                }
+            }
+
+            if let Some(part) = current {
+                if let Some(minute_str) = part.strip_suffix("M") {
+                    let minute = minute_str
+                        .parse::<i32>()
+                        .map_err(|e| ParseDurationError::from(e)
+                            .pos(pos)
+                        )?;
+                    result += (minute as i64) * MICROS_PER_MINUTE;
+                    pos += part.len();
+                    current = parts.next();
+                }
+            }
+
+            if let Some(part) = current {
+                if let Some(second_str) = part.strip_suffix("S") {
+                    let (second_str, subsec_str) = second_str
+                        .split_once('.')
+                        .map(|(sec, sub)|
+                            (sec, sub.get(..6).or_else(||Some(sub))))
+                        .unwrap_or_else(|| (second_str, None));
+
+                    let second = second_str
+                        .parse::<i32>()
+                        .map_err(|e| ParseDurationError::from(e)
+                            .pos(pos)
+                        )?;
+                    result += (second as i64) * MICROS_PER_SECOND;
+                    pos += second_str.len() + 1;
+
+                    if let Some(subsec_str) = subsec_str {
+                        let subsec = subsec_str
+                            .parse::<i32>()
+                            .map_err(|e| ParseDurationError::from(e)
+                                .pos(pos)
+                            )?;
+                        result += (subsec as i64) * 10_i64.pow(
+                            (6 - subsec_str.len()) as u32
+                        ) * if second < 0 { -1 } else { 1 };
+                        pos += subsec_str.len()
+                    }
+                    current = parts.next();
+                }
+            }
+
+            if current.is_some() {
+                Err(ParseDurationError::new("expecting EOF").pos(pos))
+            } else {
+                Ok(Self { micros: result })
+            }
+        } else {
+            Err(ParseDurationError::new("not ISO format").not_final())
+        }
+    }
+
+    fn try_from_pg_format(input: &str) -> Result<Self, ParseDurationError> {
+        enum Expect {
+            Numeric { begin: usize },
+            Alphabetic { begin: usize, numeric: i64 },
+            Whitespace { numeric: Option<i64> }
+        }
+        let mut seen = Vec::new();
+        let get_value = |start: usize, end: usize| input
```

Comment:
Prefer using `{ }` for multi-line lambdas, even if it's just single expression. Or better use `fn` here?

Reply:
Both lambdas are using `input` while `get_unit` is using the `seen` vec for duplicate checking. I'll make `get_value` a `fn`. Thanks!